### ISSUE TITLE
rvm install 2.5.1 error Error running '__rvm_make -j8',

### ISSUE THREAD

**Issue by SarahBunker** [rvm/rvm#5222]
## Description

I am trying to install a previous version of Ruby 2.5.1 and it fails. This is for a few github projects that were written in Ruby 2.5.1 because the computer I was using an older computer with that version installed while working on them. When I try to enter one of the repos written in Ruby 2.5.1 I get the error that 
```
Required ruby-2.5.1 is not installed.
To install do: 'rvm install "ruby-2.5.1"'
```
[This](https://gist.github.com/SarahBunker/5387610d3cc11eba3ea4b362e27e8b4a) is the error output from trying to install ruby 2.5.1 '

## Steps to reproduce
I installed rvm using [ubuntu instructions](https://github.com/rvm/ubuntu_rvm)
```
sudo apt-get install software-properties-common
sudo apt-add-repository -y ppa:rael-gc/rvm
sudo apt-get update
sudo apt-get install rvm
sudo usermod -a -G rvm $USER
```

I installed ruby 3.1.2 `rvm install 3.1.2` successfully.
Then I tried to install ruby 2.5.1 `rvm install 2.5.1` which caused the error.
From debugging they suggested updating rvm which I did with `rvm get stable`. This also raises an error.

Other debugging steps I have tried previously. Uninstalling rvm with `sudo apt remove rvm` and then reinstalling. I have tried installing ruby 3.0.0 as another version. That failed with similar output to ruby 2.5.1.

## Actual behavior

[This](https://gist.github.com/SarahBunker/5387610d3cc11eba3ea4b362e27e8b4a) is the error output from trying to install ruby 2.5.1 using `rvm install 2.5.1`. Installing with `rvm install "ruby-2.5.1"`  or `rvm install ruby 2.5.1` raises the same error. I am honestly not sure which one of these commands I used for the make and config logs below.
[This](https://gist.github.com/SarahBunker/94f69c62b001e9a4b612421faa8a003c) is the error from trying to update to the stable rvm.
I believe the current version is version 2, but my version is 1.19.12 which I found from running `rvm -v` which outputs `rvm 1.29.12 (manual) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io]`

### Other output requested
`config.log` returns `config.log: command not found`
`make.log` returns `make.log: command not found`

Perhaps this is something I need to install to be able to show the output?
I was able to install make using `sudo apt intall make` but the command is still not found.
I tried this with config as well using `sudo apt install config` the installer was `E: Unable to locate package config`.

The error says `please read /usr/share/rvm/log/1652919618_ruby-2.5.1/make.log`
I found a [config.log](https://gist.github.com/SarahBunker/d3635cc5b85d0170d32a82881962e449) and [make.log](https://gist.github.com/SarahBunker/78fcfaead80fccc8565cd5c3b92f1b59) file there. That is probably what the above commands are meant to output.

## Environment info

[gist](https://gist.github.com/SarahBunker/71a1928e809e81fbbdba969cca1fd9bf) for `rvm info` command. The output is also included below.

ruby-3.1.2:

  system:
    uname:        "Linux sd-MS-7A70 5.15.0-30-generic #31-Ubuntu SMP Thu May 5 10:00:34 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux"
    name:         "Ubuntu"
    version:      "22.04"
    architecture: "x86_64"
    bash:         "/usr/bin/bash => GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)"
    zsh:          " => not installed"
    remote_path:  "ubuntu/22.04/x86_64"

  rvm:
    version:      "1.29.12 (manual)"
    updated:      "2 hours 4 minutes 36 seconds ago"
    path:         "/usr/share/rvm"
    autolibs:     "[4] Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X)."

  ruby:
    interpreter:  "ruby"
    version:      "3.1.2p20"
    date:         "2022-04-12"
    platform:     "x86_64-linux"
    patchlevel:   "2022-04-12 revision 4491bb740a"
    full_version: "ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-linux]"

  homes:
    gem:          "/usr/share/rvm/gems/ruby-3.1.2"
    ruby:         "/usr/share/rvm/rubies/ruby-3.1.2"

  binaries:
    ruby:         "/usr/share/rvm/rubies/ruby-3.1.2/bin/ruby"
    irb:          "/usr/share/rvm/rubies/ruby-3.1.2/bin/irb"
    gem:          "/usr/share/rvm/rubies/ruby-3.1.2/bin/gem"
    rake:         "/usr/share/rvm/rubies/ruby-3.1.2/bin/rake"

  environment:
    PATH:         "/usr/share/rvm/gems/ruby-3.1.2/bin:/usr/share/rvm/gems/ruby-3.1.2@global/bin:/usr/share/rvm/rubies/ruby-3.1.2/bin:/usr/share/rvm/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin"
    GEM_HOME:     "/usr/share/rvm/gems/ruby-3.1.2"
    GEM_PATH:     "/usr/share/rvm/gems/ruby-3.1.2:/usr/share/rvm/gems/ruby-3.1.2@global"
    MY_RUBY_HOME: "/usr/share/rvm/rubies/ruby-3.1.2"
    IRBRC:        "/usr/share/rvm/rubies/ruby-3.1.2/.irbrc"
    RUBYOPT:      ""
    gemset:       ""

## Conclusion

Thank you for your help!

**Comment by SarahBunker** [comment]
[This](https://gist.github.com/SarahBunker/368ddd715dcc188eeed6c31a28126727) is the output.
```
...
Gemset '' does not exist, 'rvm ruby-2.5.9 do rvm gemset create ' first, or append '--create'.
```
So I don't think 2.5.9 works.

**Comment by raelgc** [comment]
Can you try again, but before it, run `rvm gemset use global`?

**Comment by SarahBunker** [comment]
That line with the output in the gist appeared after running the commands you provided above. I tried running `rvm use 2.5.9` and Instead of all of the lines of output in the gist it only had the line `Gemset '' does not exist, 'rvm ruby-2.5.9 do rvm gemset create ' first, or append '--create'`

**Comment by SarahBunker** [comment]
I just ran
```
rvm gemset use global
rvm use 2.5.9
```
and it outputted `Gemset '' does not exist, 'rvm ruby-2.5.9 do rvm gemset create ' first, or append '--create'.`

Did you mean to run
```
rvm gemset use global
cd /usr/share/rvm/rubies
wget https://raelcunha.com/share/ruby-2.5.9.tar.bz2
tar -xjvf ruby-2.5.9.tar.bz2
rvm use 2.5.9
```

**Comment by SarahBunker** [comment]
I am currently rebuilding the applications using ruby 3.1.2
I am just using it as a learning opportunity to remember how I built them.

**Comment by raelgc** [comment]
Instead of `rvm use 2.5.9.`, can you try this?

```terminal
rvm use 2.5.9@test --create --default
```

**Comment by raelgc** [comment]
Thanks for all the feedback, @SarahBunker.

The `Gemset '' does not exist, 'rvm ruby-2.5.9 do rvm gemset create ' first, or append '--create'.` error is not common, and probably not related to the issue we're trying to solve here. 

But if you can get `2.5.9` running, than I can move it to the official rvm binaries and next time it'll be a lot easier to get it working.

**Comment by Harmansharma** [comment]
Error running '__rvm_make -j4',
please read /usr/share/rvm/log/1657808807_ruby-2.5.0/make.log

There has been an error while running make. Halting the installation.


How to fix this "

**Comment by 17101233** [comment]
> Error running '__rvm_make -j4', please read /usr/share/rvm/log/1657808807_ruby-2.5.0/make.log
> 
> There has been an error while running make. Halting the installation.
> 
> How to fix this "

I am also having the same issue

**Comment by bigapple716** [comment]
Allow me to summarize on Oct 5th, 2022:
The goal is to successfully run `rvm install 2.5.1 --with-openssl-dir=path/to/openssl@1.1`. You want to use openssl v1 for Ruby v2
If you have a M1 Mac, you need to install openssl with x86 homebrew in Rosetta Terminal to get the correct version of openssl@1.1.

**Comment by kat2018** [comment]
> If you have a M1 Mac, you need to install openssl with x86 homebrew in Rosetta Terminal to get the correct version of openssl.

^ This solution have worked for me (Nov 16th, 2022).

[HERE](https://github.com/rvm/rvm/issues/5043#issuecomment-967000943) is a more detailed step by step that helped me solve the error `Error running '__rvm_make -j8'` on my MacOS Monterey with M1 chip.

**Comment by yasserakbbach** [comment]
> > If you have a M1 Mac, you need to install openssl with x86 homebrew in Rosetta Terminal to get the correct version of openssl.
> 
> ^ This solution have worked for me (Nov 16th, 2022).
> 
> [HERE](https://github.com/rvm/rvm/issues/5043#issuecomment-967000943) is a more detailed step by step that helped me solve the error `Error running '__rvm_make -j8'` on my MacOS Monterey with M1 chip.

You have just saved me, thank you.

**Comment by AndyObtiva** [comment]
@xinghengwang  Ruby has been running fine on my m1 laptop for more than a year now. This might not be related to that. I get this error on another x86_64 Intel processor laptop actually.

**Comment by BiosBoy** [comment]
Have the sane error...

**Comment by AndyObtiva** [comment]
My problem turned out to be caused by a defective installation of Homebrew (an RVM dependency on the Mac), due to missing the latest XCode.

I installed the latest XCode, reinstalled Homebrew, and reinstalled RVM, and then installing Ruby worked just fine, albeit I installed supported versions of Ruby only (2.7+).

I have RVM working on both my ARM64 Macs (M1 and M2 Chips) and X64 Mac (Intel Core i7).

I hope this solution helps you and other people facing the problem.
